File: Lib/Kvaser/Canlib/Samples/NET/vs2010/CPlusPlusEvent/CanlibWaitEvent.cpp
#include "StdAfx.h"
#include "CanlibWaitEvent.h"

using namespace System;
using namespace CPlusPlusEvent;
using namespace Microsoft::Win32::SafeHandles;

CanlibWaitEvent::CanlibWaitEvent(Object^ we)
{
	unsigned int theHandle = (unsigned int)we;
	IntPtr pointer = IntPtr((Int64)theHandle) ;
	Microsoft::Win32::SafeHandles::SafeWaitHandle^ swHandle = 
		gcnew Microsoft::Win32::SafeHandles::SafeWaitHandle(pointer, true) ;
	SafeWaitHandle = swHandle;
}

File: Lib/Kvaser/Canlib/Samples/NET/vs2010/VBEvent/CanlibWaitEvent.vb
Friend Class CanlibWaitEvent
    Inherits System.Threading.WaitHandle

    Public Sub New(ByVal we As Object)
        Dim theHandle As UInt32 = DirectCast(we, UInt32)
        Dim pointer As New IntPtr(CLng(theHandle))
        Dim swHandle As New Microsoft.Win32.SafeHandles.SafeWaitHandle(pointer, True)
        MyBase.SafeWaitHandle = swHandle
    End Sub

End Class
